src/ucx_van.h

```diff
@@ -482,6 +482,7 @@ class UCXVan : public Van {
     CHECK(IsDataMsg(msg));
 
     ucp_tag_t tag       = MakeTag(my_node_.id, UCX_TAG_DATA);
+    tag = (tag << 32) | msg.meta.key;
```

Comment:
this way you discard node_id from the tag on both sender and receiver side. 
then it is possible that server will receive a message from the wrong worker

Reply:
the `node_id` is still stored in the high 32 bits of `tag`. Maybe a better way for L485 is 
```tag = (tag << 32) | (msg.meta.key & 0x0000FFFF);```

Reply:
`MakeTag` saves nodeId in high 32 bits. Line 485 replaces nodeId with  `UCX_TAG_DATA` and set meta.key instead of it.

Reply:
I see. Seems like there's no very clean way to encode the node_id, UCX_TAG_DATA and msg.meta.key in the same tag. this should work in most cases:
```
// UCX_TAG_DATA = 0x8000
tag = my_node_.id | UCX_TAG_DATA;
tag = (tag << 32) | (msg.meta.key & 0x0000FFFF);
```

do you have suggestions to make sure tag_send/tag_recv match up correctly other than using a different tag for each pair?

Reply:
@brminich do you have concerns if we change L485 to the form in my last comment?

Reply:
this would shorten the key to 32 bytes (btw, need to use 0xffffffff mask). Is it ok?
I'm looking on how to support full 8-bytes key

Reply:
yes the lower 32 bits of a `msg.meta.key` is still unique within each worker. yes the mask should be `0xffffffff`. If we combine the lower 32 bits from this key with the 32 bits node_id into the tag, that would make all tags unique. In practice there won't be so many worker nodes to use all bits of `node_id`, it should be safe to use the highest bit of node id as the `UCX_TAG_DATA/UCX_TAG_META` bit.

Reply:
then such change seems to be ok
👍